Classes/Integration/FormEngine/ColposElement.php

```diff
@@ -0,0 +1,101 @@
+<?php
+namespace FluidTYPO3\Flux\Integration\FormEngine;
+
+/*
+ * This file is part of the TYPO3 CMS project.
+ *
+ * It is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License, either version 2
+ * of the License, or any later version.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ *
+ * The TYPO3 project - inspiring people to share!
+ */
+
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Backend\Form\Element\AbstractFormElement;
+
+use FluidTYPO3\Flux\Provider\Interfaces\GridProviderInterface;
+use FluidTYPO3\Flux\Service\FluxService;
+use FluidTYPO3\Flux\Utility\ColumnNumberUtility;
+use TYPO3\CMS\Core\Database\ConnectionPool;
+use TYPO3\CMS\Extbase\Object\ObjectManager;
+use TYPO3\CMS\Extbase\Utility\LocalizationUtility;
+use TYPO3\CMS\Core\Utility\StringUtility;
+
+
+/**
+ * Generation of elements of the type "user"
+ */
+class ColposElement extends AbstractFormElement
+{
+    /**
+     * User defined field type
+     *
+     * @return array As defined in initializeResultArray() of AbstractNode
+     */
+    public function render()
+    {
+        $resultArray = $this->initializeResultArray();
+
+        $parameters = $this->data['parameterArray'];
+        $colPos = $parameters['itemFormElValue'];
+        $inputValue = (string) $colPos;
+
+        $id = StringUtility::getUniqueId('formengine-flux-colPos-');
+        $resultArray['html'] = sprintf(
+            '<input type="hidden" name="%s" id="%s" class="flux-flex-colPos-input" value="%s" />Column position: <strong class="flux-flex-colPos-text">%d</strong>',
+            $parameters['itemFormElName'],
+            $id,
+            $inputValue,
+            $inputValue
+        );
+
+        if ($inputValue === '') {
+            $rowUid = $this->data['databaseRow']['uid'];
+            if (!isset($rowUid) || !is_int($rowUid)) {
+                $rowUid = 0;
+            }
+            $usedColumnPositions = $this->determineUsedColumnPositionsWithinParent($this->data['tableName'], $rowUid);
+            $usedColumnPositionsJS = '[' . implode(',', $usedColumnPositions) . ']';
+            // FIXME better constant?
+            $maximumColPos = 99;
+
+            $resultArray['additionalJavaScriptPost'][] = sprintf("(function() {
```

Comment:
If possible, the JS should be in a RequireJS module so we only load the module. The JS should then read all of the necessary variables from DOM properties of our hidden field. There is an existing such module loaded by the `Preview` class which interacts with our grids - perhaps you can use that as reference (and somehow make it also listen for newly added DOM nodes instead of just running once after page load).

Reply:
> The JS should then read all of the necessary variables from DOM properties of our hidden field.

OK, changing the data to DOM properties should be easy.

> (and somehow make it also listen for newly added DOM nodes instead of just running once after page load)

This does not run after page load, it runs each time you add a new panel.

Reply:
> This does not run after page load, it runs each time you add a new panel.

This was about the grid require-js module, not the code you wrote. It needs to behave like you wrote it, but should happen with listeners instead of ad-hoc execution in an inserted script block ;)

Reply:
Ah, I misread the context. Thanks for clarifying!